enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,493 @@
+example is write permission on an S3 bucket on AWS. STS is supported on all the major cloud providers where OpenShift is
+deployed. Currently, several Red Hat operators provide detailed instructions for how to use the operator on STS-enabled
+clusters. The process is designed to have manual operations performed by the customer. This enhancement seeks to bring common standards to the process of enabling an
+operator to access cloud provider resources, by enabling automation, UX and logic for the process. 
+
+## Motivation
+
+Several important Red Hat operators have normal operating modes where they assure their operands have required cloud 
+resources to perform properly. Currently, when a cluster is installed on a cloud platform an admin is required to 
+follow instructions (for example [1]) to provide STS-based credentials to the operator such that needed resources can be
+accessed by the operand. This enhancement seeks to normalize this process across operators, so users of several of them
+have the same experience and similar steps to perform. We aim to automate and reduce those steps as much as possible to 
+make adding operators (via OLM) simpler on an STS enabled cluster.
+
+### User Stories
+
+* As a cluster admin I know which OLM Operators support tokenized authentication for my cloud.
+* As a cluster admin of a cluster using tokenized cloud auth I know what's required to install and upgrade OLM Operators whenever those operators manage resources that authenticate against my cloud.
+* As an Operator developer, I have a standard framework to define tokenized authentication requirements and consume them, per supported cloud.
+* As an Operator Hub browser I know which operators support tokenized cloud auth and on which clouds.
+* As the Hypershift team, where the only credential mode for clusters/customers is STS (on AWS) , the Red Hat branded
+  Operators that must reach the AWS API, should be enabled to work with STS credentials in a consistent, and automated
+  fashion that allows customer to use those operators as easily as possible, driving the use of layered products.
+
+### Goals
+
+Allow Day-2 Operators to access cloud provider resources as seamlessly as possible when being installed, used and
+updated on STS enabled clusters.
+
+While providing the above goal, allow for multi-tenancy. In this sense multi-tenancy means that an operator may
+provide different cloud provider credentials per operand such that operator users may uniquely access non-shared cloud
+resources.
+
+Operator authors have a way to notify, guide, and assist Day-2 Operator admins in providing the required cloud provider
+credentials matched to their permission needs for install and update.
+
+Ideally, a solution here will work in both HyperShift (STS always) and non-HyperShift clusters.
+
+### Non-Goals
+
+Day 1 operators are not included in this proposal, they are pre-provisioned for STS enabled clusters (at least 
```

Comment:
So for an IPI cluster, the user still needs to set up the STS tokens manually for the core, then gets day 2 automation? That seems like an odd choice to me, is there a strong issue with changing the core which makes it expand the scope of this enhancement drastically

Reply:
There's still setup for the day 2 operators.

My understanding is that ROSA *might* automate the creation of roles and attach the appropriate ARN to the olm subscription, but that's only ROSA and only if they have the permissions to be able to do that.  (This is dependent on a few things, not the least of which is coming up with a way for the olm operator to communicate what roles/permissions it needs, in a programmatically, and bundle-version-specific, way)

Anyone installing an operator not on ROSA, or not going through the ROSA cli, is going to have to setup the aws roles manually based on guidance from the olm operator about what it needs.

Reply:
I think you asked and answered. Time. If we expand this scope to include more than Day 2 operators we increase the time needed to complete this.

Reply:
Some additional context:  The focus here is on ROSA STS clusters, not STS clusters in general(though they may benefit from improved workflows here).

The concern is that it currently looks strange to customers that when they provision a service(ROSA) through AWS(or through the rosa command line), that service requires the user go do additional things elsewhere in AWS.  So the IPI scenario is not the same and doesn't trigger the same sort of disjoint/schizophrenic UX.

Reply:
@JoelSpeed I don't see why core operators can't do this in the future but its time we're restricted to at the moment, this feature shouldn't prevent us from automating the secret creation for core operators in the future. We will still have the issue of getting the ARN to the CredentialsRequest for core operators which will likely require changes in the installer to better assist users. The focus here is day-2.

Reply:
I think this can be marked resolved now